cheri_compressed_cap_common.h

```diff
@@ -580,9 +649,12 @@ static inline bool _cc_N(is_representable_new_addr)(bool sealed, _cc_addr_t base
         c.cr_otype = sealed ? 42 : _CC_N(OTYPE_UNSEALED); // important to set as compress assumes this is in bounds
         /* Get an EBT */
         bool exact_input = false;
-        c.cr_ebt = _cc_N(compute_ebt)(base, top, NULL, &exact_input);
-        _cc_debug_assert(exact_input &&
-                         "Input capability bounds not representable? They should have been rounded before!");
+        c.cached_pesbt = _cc_N(recompute_pesbt_non_ebt)(&c);
+        _cc_N(update_ebt)(&c, _cc_N(compute_ebt)(base, top, NULL, &exact_input));
+        // Looks like this assert gets hit by negative length capabilities. Probably the "exact input" return is wrong.
+        if (top > (_cc_length_t)base)
+            _cc_debug_assert(exact_input &&
```

Comment:
Would be good to fix this return value instead of working around it. But if you open an issue we can fix this after initial QEMU support.

Reply:
Agreed.